filesystems/blob/src/test/scala/cromwell/filesystems/blob/BlobPathBuilderSpec.scala

```diff
@@ -47,17 +62,66 @@ class BlobPathBuilderSpec extends AnyFlatSpec with Matchers{
   }
 
   ignore should "build a blob path from a test string and read a file" in {
-    val endpoint = BlobPathBuilderSpec.buildEndpoint("coaexternalstorage")
-    val endpointHost = BlobPathBuilder.parseURI(endpoint).getHost
+    val storageAccountName = "coaexternalstorage"
+
+    val profile = new AzureProfile(AzureEnvironment.AZURE)
+    val azureCredential = new DefaultAzureCredentialBuilder()
+      .authorityHost(profile.getEnvironment.getActiveDirectoryEndpoint)
+      .build
+    val azure = AzureResourceManager.authenticate(azureCredential, profile).withDefaultSubscription
+
+    // TODO: Can we get the storage account key without all this java-to-scala conversion?
+    val storageAccounts = azure.storageAccounts()
+    val storageAccount = storageAccounts
+      .list()
+      .asScala
+      .find(_.name == storageAccountName)
+      .asInstanceOf[Some[StorageAccount]]
+      .get
+    val storageAccountKeys = storageAccount.getKeys()
+      .asScala
+      .map(f => f.value)
+    val storageAccountKey = storageAccountKeys.head
+
+    val keyCredential = new StorageSharedKeyCredential(
+      storageAccountName,
+      storageAccountKey
+    )
+    val endpoint = BlobPathBuilderSpec.buildEndpoint(storageAccountName)
+    val blobServiceClient = new BlobServiceClientBuilder()
+      .credential(keyCredential)
+      .endpoint(endpoint)
+      .buildClient()
+
+    val accountSasPermission = new AccountSasPermission()
+      .setReadPermission(true)
+    val services = new AccountSasService()
+      .setBlobAccess(true)
+      .setFileAccess(true)
+      .setQueueAccess(true)
+      .setTableAccess(true)
+    val resourceTypes = new AccountSasResourceType()
+      .setObject(true)
+      .setService(true)
+      .setContainer(true)
```

Comment:
We'd like these to be as narrow as possible, can you try a few things here and see if you can find a minimum set of permissions that will work to open the filesystem?